src/components/modal/ModalSelect.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button, Icon, Text, View } from 'native-base';
+import React from 'react';
+import Modal from 'react-native-modal';
+import computeFormStyleSheet from '../../FormStyles';
+import BaseProps from '../../types/BaseProps';
+import Utils from '../../utils/Utils';
+import { ItemSelectionMode } from '../list/ItemsList';
+import computeStyleSheet from './ModalStyles';
+import I18n from 'i18n-js';
+import SelectableList from '../../screens/base-screen/SelectableList';
+
+export interface Props<T> extends BaseProps {
+  defaultItem?: T;
+  buildItemName: (item: T) => string;
+  selectionMode: ItemSelectionMode;
+  onItemsSelected: (selectedItems: T[]) => void;
+}
+interface State<T> {
+  isVisible: boolean;
+  selectedItems: T[];
+  canValidateMultiSelection: boolean;
```

Comment:
No because selectedItems will always have the default item inside, even when no items has been selected yet

Reply:
In that case we should not mix selectedItems with defaultItems.

Reply:
Yes I thought about it and I know how to change it

Reply:
Ok, even without mixing the two, the boolen is needed because when opening the modal, the selectedItems won't be empty because it will hold the last selected items (that are required inside the modal to handle the (+n) indication as well as the 'validate' button activation)

Reply:
Yes, the selected items must be emptied whenever the modal is displayed.
You have componentDidFocus/Blur methods to handle this use case.